src/components/Collections/components/ContestCollections/ContestCollectionItemScorer.tsx

```diff
@@ -0,0 +1,67 @@
+import { Button, SegmentedControl, Text } from '@mantine/core';
+import { useState } from 'react';
+import { useSetCollectionItemScore } from '~/components/Collections/collection.utils';
+
+export function ContestCollectionItemScorer({
+  collectionItemId,
+  currentScore = 0,
+  onScoreChanged,
+  layout = 'default',
+}: {
+  layout?: 'default' | 'minimal';
+  collectionItemId: number;
+  currentScore?: number;
+  onScoreChanged?: (data: { userId: number; collectionItemId: number; score: number }) => void;
+}) {
+  const [selectedScore, setSelectedScore] = useState(currentScore);
+
+  const { setItemScore, loading } = useSetCollectionItemScore();
+  const handleSetItemScore = async (_score?: number) => {
+    await setItemScore(
+      {
+        collectionItemId,
+        score: _score ?? selectedScore,
+      },
+      {
+        onSuccess: (result) => {
+          if (onScoreChanged)
+            onScoreChanged({
+              userId: result.userId,
+              collectionItemId: result.collectionItemId,
+              score: result.score,
+            });
+        },
+      }
+    );
```

Comment:
Ditto, we should probably catch the error here, so it doesn't bubble up to the client app